### ISSUE TITLE
Возможность управления правами пользователя для систем на базе IOS XR

### ISSUE THREAD

**Issue by zirus00** [tacacsgui/tacacsgui#6]
Для IOS XR сейчас в ручном режиме приходится заносить настройки в конфигурацию:
service = exec { optional task = "#root-system" }
или
service = exec { task = "rwx:bgp,#operator" }

Параметр rwx:bgp означает доступ соответствующему пользователю на read/write/execute по сервису (в терминологии Cisco - task) BGP, а с решетки начинаются названия taskgroup которые настраиваются на роутере и пользователь назначается в соответствующую группу где уже прописаны права по всем таскам (rwx).

Ключевое слово "optional" требуется для работы такакса с легаси Cisco IOS устройствами, без него пользователь не может авторизоваться на IOS устройстве, тк роутер не понимает нового атрибута

Суть реквеста в просьбе реализовать  возможность управления правами пользователей на IOS XR

**Comment by zirus00** [comment]
> Мой вопрос: использует ли iOS XR service = shell ... я хочу придумать, как это правильнее реализовать.

Дернул такакс, в packet capture вижу запрос на авторизацию, роутер запрашивает
service=shell
cmd*
task*

**Comment by gurkin33** [comment]
К сожалению, пока что сервис надо прописывать вручную, я не могу представить как это реализовать в интерфейсе, т.к. у меня не достаточно информации по все устройством, которые могут использовать сервис.
Я буду вам очень признателен, если вы дадите пример конфигурации любых устройств, а если еще и атрибуты сервисов будут (не стандартных), благодарности моей не будет предела. 
Так же хочу отметить, что если вам нужно написать, что-то типа service = exec или service = juniper, то вы можете использовать полностью ручную настройку. Если найдётся случай, когда надо добавлять несколько сервисов, я реализую это, а пока что так.

С Уважением, Алексей

**Comment by zirus00** [comment]
Да, я как раз сейчас использую ручную настройку, понятно что все сразу предусмотреть невозможно, я тестирую, проверяю на железе которое у меня есть, Cisco IOS/IOS XE, IOS XR, Huawei VRP, Cisco sg300 - все используют сервис shell, Cisco WLC использует сервис ciscowlc, про juniper я знаю только из интернета, информацию по всем атрибутам приходится смотреть в прямо в пакетах Tacacs через wireshark.

Я обязательно отпишусь по всем результатам тестов

**Comment by gurkin33** [comment]
Так же есть вопрос по поводу перехвата пакетов, вы используете span потр для перехвата пакетов? Или tcpdump и потом wireshark?) Если честно, я не доходил до такого дебага, хотелось бы перенять опыт.

**Comment by zirus00** [comment]
Я делаю захват трафика прямо на такакс сервере и передаю его на локальный wireshark
`ssh tacacs "tcpdump -s0 -U -n -w -  'port 49'" | wireshark -k -i -`
Потом в wireshark указываю ключ который настроил и он все дешифрует, сразу в запросах на авторизацию видно и сервис и параметры которые она запрашивает

**Comment by gurkin33** [comment]
Добрый день!

Наконец-то дошел и до этого есть у меня одна идея, но мне нужна Ваша помощь. Могли бы вы написать мне в личную почту? Спасибо!

С Уважением, Алексей
